flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,168 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.api.common.restartstrategy.RestartStrategies;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.api.core.Pipeline;
+import org.apache.flink.ml.api.core.Stage;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.ExecutionCheckpointingOptions;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/** knn algorithm test. */
+public class KnnTest {
```

Comment:
Could these unit tests verify the numerical correctness of the outputs, instead of simply doing System.out?

In general we don't do System.out in unit tests in Flink. The existing `KMeansTest` and the `NaiveBayes` PR is able to test numerical correctness. And I think Spark also has numerical correctness test for most algorithms.

Reply:
OK, I will refine the ut later.

Reply:
done

Reply:
public void invoke(Row value, Context context) {
                                String label = (String) value.getField(0);
                                String pred = (String) value.getField(1);
                                assert (label.equals(pred));
    }

in my ut, I assert the predict results is wright. 
knn is a classification algirithm, I user a train data train the model, use this model to predict a predData, and check the correctness of the result.